### ISSUE TITLE
The v0.15.10 failed to parse the 407 message

### ISSUE THREAD

**Issue by portsip** [onsip/SIP.js#792]
**Describe the bug**
We have our own WebRTC gateway, it works fine with the v0.78 SIP.JS.

But with the v0.15.10, when WebRTC client(base on v0.15.10) registers to our gateway, the gateway sends back "407 Proxy Authentication Required" to the client,  it seems the client can't parse it (with v0.78 it works perfectly), the finishData is empty after converting binary data to a string, so cause the CLRF can't be found.

As we debugged, before convert to string, in v0.78,  SIP.JS uses the ArraryBuffer to store the binary data, in v0.15.10, the binary data is store in a BLOB, this is the difference we found.

**Logs**
I've attached the screenshot of the client debug console, below is the WebRTC gateway log, include the 407 message sent to the client.
In the screenshot, we can see the data size is 506 bytes, but the "finishData" is empty.

```

DEBUG | 20200423-115959.748 | WebRtcGW | RESIP:TRANSPORT | 7400 | transportselector.cxx:1335 | Transmitting to [ V4 218.76.62.167:11103 WSS flowKey=1476 transportKey=4 ] tlsDomain= via [ V4 172.16.232.164:10443 WSS flowKey=1476 transportKey=4 ]

SIP/2.0 407 Proxy Authentication Required
Via: SIP/2.0/WSS mrpr0n90ehp7.invalid;branch=z9hG4bK6413178;received=218.76.62.167
Proxy-Authenticate: Digest nonce="13232087999:86cab2a33203df81daa66c3ea9807da2",algorithm=MD5,realm="sip.webrtc.site"
To: <sip:103@sip.webrtc.site>;tag=3fa7c62a
From: "SIP.js Demo" <sip:103@sip.webrtc.site>;tag=nlnshcfirk
Call-ID: vqearidh0vnpdc1nm5ab
CSeq: 3 REGISTER
Server: PortSIP WebRTCGateway 12.2.0
User-Agent: PortSIP UC - Call Manager 12.1.0
Content-Length: 0


```
**To Reproduce (if possible)**

1. Open https://sip.webrtc.site:8881/demo2/demo-1.html
2. Click "connect", will connect to WSS server , and send register message.
3.  WebRTC Gateway received the register message, and replies 407 message.
4.  Since the finishData is empty, so can't find the CRLF, the demo client discard the message - please see attached the screenshot. Again, the v0.78 works fine with the same server and the same account.
5. Test with v0.78:
  
>     Open https://collecttix.github.io/ctxSip/
>     Click "Try the Demo" will pop up a window, enter below details:
>       user: 101, 102
>       password: 111111
>       wss server: wss://sip.webrtc.site:10443
>       realm: sip.webrtc.site
> 
>    After signing, you can make the call between 101 and 102

**Expected behavior**
Expect the WebRTC client parse the data correctly.

**Observed behavior**
The browser doesn't parse the 407 message correctly.

**Environment Information**
 - Backend Server and version: PortSIP WebRTCGateway 12.2.0
 - Browser and version: Chrome 81.0.4044.122 (64bit)

**Additional context**

With the same environment, the SIP.JS v0.78 works fine, but we would like to update it to the new version.

![sipjs](https://user-images.githubusercontent.com/11405486/80060739-ddef7080-8561-11ea-8ac6-91e0a44aecf9.jpg)

**Comment by john-e-riordan** [comment]
I was able to reproduce with your test case (thanks), fix the issue, and re-test against your server to confirm the fix works. It has apparently been broken since 0.11.0. I believe my fix will get pulled in and go out in the next 0.15.x release.

**Comment by portsip** [comment]
Thank you, I will apply your fix to test!.
Have a nice day!

**Comment by portsip** [comment]
After we applied your fix, it works fine now. 
Thanks a lot!

**Comment by james-criscuolo** [comment]
This has been merged and release with `0.15.11`, closing. Thanks again!

**Comment by alisherafat01** [comment]
How to solve 407 in 0.21 version of `sip.js`? where is the place to provide password for `AOR`?
